src/transformers/models/detr/modeling_detr.py

```diff
@@ -648,9 +648,10 @@ def forward(
         hidden_states = residual + hidden_states
         hidden_states = self.final_layer_norm(hidden_states)
 
-        if torch.isinf(hidden_states).any() or torch.isnan(hidden_states).any():
-            clamp_value = torch.finfo(hidden_states.dtype).max - 1000
-            hidden_states = torch.clamp(hidden_states, min=-clamp_value, max=clamp_value)
+        if self.training:
+            if torch.isinf(hidden_states).any() or torch.isnan(hidden_states).any():
+                clamp_value = torch.finfo(hidden_states.dtype).max - 1000
+                hidden_states = torch.clamp(hidden_states, min=-clamp_value, max=clamp_value)
```

Comment:
@stas00 IS that ok to remove at inference time ?

Reply:
In theory yes. In practice, it depends on how the model was pre-trained.

The model weights don't change during inference, so we don't need to keep things in check all the time.

However if the pre-trained model's weights lead to an overflow in a single iteration during training, as is the case with some mt5 models under mixed-precision then this can occur just as well during inference.

This is primarily an issue with pre-trained on bf16 models fine-tuned/inferenced on fp16 (mixed or non-mixed precision).

If a model was pretrained with fp16/mixed precision it's pretty sure the clamping won't be needed.

To give you a more intelligent answer it'd require running some tests with the actual DETR models and checking its activations magnitudes, which should be pretty trivial, using https://huggingface.co/transformers/debugging.html#underflow-and-overflow-detection, which can be plugged into HF Trainer and the examples.

Reply:
To be honest I think this code was just badly copy pasted, so I'm more in favor of disabling this hack for training (as it is done now)

Reply:
Ok, if everyone is favorable, then let's do this.

Reply:
> To be honest I think this code was just badly copy pasted, so I'm more in favor of disabling this hack for training (as it is done now)

you must have meant for inference, right Patrick?